Stratis.Bitcoin/Wallet/WalletManager.cs

```diff
@@ -0,0 +1,918 @@
+        /// <param name="isChange">Whether the addresses added are change (internal) addresses or receiving (external) addresses.</param>
+        /// <returns>A list of addresses in Base58.</returns>
+        private List<string> CreateAddressesInAccount(HdAccount account, Network network, int addressesQuantity, bool isChange = false)
+        {
+            List<string> addressesCreated = new List<string>();
+
+            var addresses = isChange ? account.InternalAddresses : account.ExternalAddresses;
+
+            // gets the index of the last address with transactions
+            int firstNewAddressIndex = 0;
+            if (addresses.Any())
+            {
+                firstNewAddressIndex = addresses.Max(add => add.Index) + 1;
+            }
+
+            for (int i = firstNewAddressIndex; i < firstNewAddressIndex + addressesQuantity; i++)
+            {
+                // generate new receiving address
+                BitcoinPubKeyAddress address = this.GenerateAddress(account.ExtendedPubKey, i, isChange, network);
+
+                // add address details
+                addresses.Add(new HdAddress
+                {
+                    Index = i,
+                    HdPath = CreateBip44Path(account.GetCoinType(), account.Index, i, isChange),
+                    ScriptPubKey = address.ScriptPubKey,
+                    Address = address.ToString(),
+                    Transactions = new List<TransactionData>()
+                });
+
+                addressesCreated.Add(address.ToString());
+            }
+
+            if (isChange)
+            {
+                account.InternalAddresses = addresses;
+            }
+            else
+            {
+                account.ExternalAddresses = addresses;
+            }
+
+            return addressesCreated;
+        }
+
+        /// <inheritdoc />
+        public Wallet GetWallet(string walletName)
+        {
+            Wallet wallet = this.GetWalletByName(walletName);
+            return wallet;
+        }
+
+        /// <inheritdoc />
+        public IEnumerable<HdAccount> GetAccountsByCoinType(string walletName, CoinType coinType)
+        {
+            Wallet wallet = this.GetWalletByName(walletName);
+
+            return wallet.GetAccountsByCoinType(coinType);
+        }
+
+	    public int LastBlockHeight()
+	    {
+		    if (!this.Wallets.Any())
+		    {
+			    return this.chain.Tip.Height;
+		    }
+
+			return this.Wallets.Min(w => w.AccountsRoot.Single(a => a.CoinType == this.coinType).LastBlockSyncedHeight) ?? 0;
+	    }
+
+	    /// <inheritdoc />
+		public List<UnspentInfo> GetSpendableTransactions()
+	    {
+		    var outs = new List<UnspentInfo>();
+		    var accounts = this.Wallets.SelectMany(wallet => wallet.AccountsRoot.Single(a => a.CoinType == this.coinType).Accounts);
+
+		    foreach (var account in accounts)
+		    {
+			    foreach (var externalAddress in account.ExternalAddresses)
```

Comment:
can reduce this to 1 foreach by using this.ExternalAddresses.Concat(this.InternalAddresses);